packages/neon/neon_talk/lib/src/widgets/rich_object/file.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:file_icons/file_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:neon_framework/blocs.dart';
+import 'package:neon_framework/theme.dart';
+import 'package:neon_framework/utils.dart';
+import 'package:neon_framework/widgets.dart';
+import 'package:nextcloud/core.dart';
+import 'package:nextcloud/spreed.dart' as spreed;
+
+/// Displays a file preview from a rich object.
+class TalkRichObjectFile extends StatelessWidget {
+  /// Creates a new Talk rich object file.
+  const TalkRichObjectFile({
+    required this.parameter,
+    required this.textStyle,
+    super.key,
+  });
+
+  /// The parameter to display.
+  final spreed.RichObjectParameter parameter;
+
+  /// The TextStyle to applied to all text elements in this rich object.
+  final TextStyle? textStyle;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget child;
+
+    if (parameter.previewAvailable == spreed.RichObjectParameter_PreviewAvailable.yes) {
+      final maxHeight = MediaQuery.sizeOf(context).height ~/ 2;
```

Comment:
```suggestion
      final maxHeight = MediaQuery.sizeOf(context).height / 2;
```

We do not need to have it as an integer.

Reply:
Not at that point, but later on we only need the integer. It's easier and more performant to convert the double to integer as early as possible.

Reply:
Why do we need the maxHeight as an integer?
In lines 45-46 a difference of op to `0.5` wouldn't be that significant.

And further down the maxHeight is used as a double